### ISSUE TITLE
Oxlint language server panics in disable_fix.rs while using oxlint-tailwindcss

### ISSUE THREAD

**Issue by frixou89** [sergioazoc/oxlint-tailwindcss#106]
I don't know if I should open this issue here or in oxc repo but this only happens when I use this plugin... 

## Description

The Oxlint language server repeatedly panics while linting a project that uses `oxlint-tailwindcss`.

The panic occurs inside Oxlint's fixer implementation:

```text
crates/oxc_linter/src/fixer/disable_fix.rs:52:22
range end index 4648 out of range for slice of length 0
```

After the panic, the language server exits with `SIGABRT`, restarts, and immediately crashes again.

## Expected behavior

The plugin should report any applicable diagnostics without causing the Oxlint language server to panic.

If a fix or diagnostic range is invalid, it should be ignored or reported as an error rather than crashing the entire language server.

## Actual behavior

The language server panics with an out-of-bounds slice error and repeatedly restarts:

```text
thread '<unnamed>' panicked at crates/oxc_linter/src/fixer/disable_fix.rs:52:22:
range end index 4648 out of range for slice of length 0

Server process exited with signal SIGABRT.
Connection to server got closed. Server will restart.
```

## Environment

* **Oxlint:** `1.77.0`
* **Oxlint run mode:** `onType`
* **disableNestedConfig:** `false`
* **oxlint-tailwindcss version:** `1.7.0`
* **Operating system:** `Arch Linux`
* **Bun version:** `1.3.14`

## Logs

<details>
<summary>Full language server logs</summary>

```text
2026-08-06 16:08:13.144 [error]

2026-08-06 16:08:13.144 [error] thread '<unnamed>' (440401) panicked at crates/oxc_linter/src/fixer/disable_fix.rs:52:22:
2026-08-06 16:08:13.144 [error] range end index 4648 out of range for slice of length 0
2026-08-06 16:08:13.144 [error] note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
2026-08-06 16:08:41.209 [error] Server process exited with signal SIGABRT.
2026-08-06 16:08:41.210 [info] Connection to server got closed. Server will restart.
2026-08-06 16:08:41.253 [error] 2026-08-06T13:08:41.253081Z INFO oxc_language_server::backend: initialize: Some([WorkspaceOption { workspace_uri: Uri(Uri { scheme: "file", authority: Some(Authority { userinfo: None, host: "", host_parsed: RegName(""), port: None }), path: "<...>", query: None, fragment: None }), options: Object {"disableNestedConfig": Bool(false), "run": String("onType"), "flags": Object {"disable_nested_config": String("false")}} }])
2026-08-06 16:08:41.253 [error] 2026-08-06T13:08:41.253097Z INFO oxc_language_server::backend: oxlint version: 1.77.0
2026-08-06 16:08:41.510 [error]

2026-08-06 16:08:41.510 [error] thread '<unnamed>' (442360) panicked at crates/oxc_linter/src/fixer/disable_fix.rs:52:22:
2026-08-06 16:08:41.510 [error] range end index 4648 out of range for slice of length 0
2026-08-06 16:08:41.510 [error] note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
2026-08-06 16:09:09.396 [error] Server process exited with signal SIGABRT.
2026-08-06 16:09:09.404 [info] Connection to server got closed. Server will restart.
2026-08-06 16:09:09.456 [error] 2026-08-06T13:09:09.452356Z INFO oxc_language_server::backend: initialize: Some([WorkspaceOption { workspace_uri: Uri(Uri { scheme: "file", authority: Some(Authority { userinfo: None, host: "", host_parsed: RegName(""), port: None }), path: "<...>", query: None, fragment: None }), options: Object {"disableNestedConfig": Bool(false), "run": String("onType"), "flags": Object {"disable_nested_config": String("false")}} }])
2026-08-06 16:09:09.456 [error] 2026-08-06T13:09:09.452377Z INFO oxc_language_server::backend: oxlint version: 1.77.0
2026-08-06 16:09:09.708 [error]

2026-08-06 16:09:09.708 [error] thread '<unnamed>' (443161) panicked at crates/oxc_linter/src/fixer/disable_fix.rs:52:22:
2026-08-06 16:09:09.708 [error] range end index 4648 out of range for slice of length 0
2026-08-06 16:09:09.708 [error] note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

</details>

**Comment by frixou89** [comment]
I see that a similar issue was reported on [https://github.com/oxc-project/oxc/issues/25278](https://github.com/oxc-project/oxc/issues/25278)

**Comment by sergioazoc** [comment]
Context for anyone who lands here: **this is not a plugin bug, and it isn't fixed yet.**

The panic comes from a regression in **oxlint 1.77.0 itself** — `crates/oxc_linter/src/fixer/disable_fix.rs` was added on 2026-07-29, after 1.76.0, and 1.77.0 is the first release carrying it. `run_external_rules()` takes the `semantic` to run JS-plugin rules and leaves the host context's empty, so when the language server asks for ignore fixes, `add_ignore_fix()` indexes into a zero-length slice.

That means **any diagnostic from any JS plugin** triggers it — the upstream report reproduces with `@stylistic/eslint-plugin`, not with this one. You hit it through us simply because we're the plugin producing diagnostics in your project.

- Upstream issue: https://github.com/oxc-project/oxc/issues/25278 (open)
- Fix: https://github.com/oxc-project/oxc/pull/25280 (open, unreleased)

**Workaround:** pin `oxlint@1.76.0` while you use the editor extension. The CLI never enables ignore fixes, so `oxlint` / `oxlint --fix` in the terminal and in CI are unaffected on 1.77.0 — only the language server crashes.

Documented in the README, the setup guide, and the changelog for the upcoming release. I'll drop the note once a version of oxlint ships the fix.
